packages/astro/src/transitions/router.ts

```diff
@@ -13,9 +13,14 @@ type Events = 'astro:page-load' | 'astro:after-swap';
 // only update history entries that are managed by us
 // leave other entries alone and do not accidently add state.
 const persistState = (state: State) => history.state && history.replaceState(state, '');
-export const supportsViewTransitions = !!document.startViewTransition;
+
+const inBrowser = import.meta.env.SSR === false;
+
+export const supportsViewTransitions = inBrowser && !!document.startViewTransition;
+
 export const transitionEnabledOnThisPage = () =>
-	!!document.querySelector('[name="astro-view-transitions-enabled"]');
+    supportsViewTransitions && !!document.querySelector('[name="astro-view-transitions-enabled"]');
```

Comment:
```suggestion
    inBrowser && !!document.querySelector('[name="astro-view-transitions-enabled"]');
```

Reply:
Wouldn't this skip the check for `'startViewTransition' in document`? Is that intentional?

Reply:
The original code set `transitionEnableOnThisPage` to `!!document.querySelector('[name="astro-view-transitions-enabled"]')` independent of the existence of `document.startViewTransition` 
The new code should do the same (but only in the browser)